src/System.Text.RegularExpressions/tests/Regex.Split.Tests.cs

```diff
@@ -50,7 +50,8 @@ public static IEnumerable<object[]> Split_TestData()
         }
 
         [Theory]
-        [MemberData(nameof(Split_TestData))]
+        [MemberData(nameof(Split_NonCompiled_TestData))]
+        [MemberData(nameof(RegexCompilationHelper.TransformRegexOptions), nameof(Split_NonCompiled_TestData), 2, MemberType = typeof(RegexCompilationHelper))]
```

Comment:
This TransformRegexOptions scheme is an elaborate way of basically doing a simple thing, |-ing `RegexOptions.Compiled`. It took me a while to figure it out, and what regexOptionsArrayIndex is, etc. Can we do it without reflection? eg maybe update the generators:

```c#
[Theory]
[MemberData(nameof(Split_NonCompiled_TestData), nameof(enableCompilationFlag)]
[MemberData(nameof(Split_NonCompiled_TestData), nameof(noCompilationFlag)]
...
```
and
```c#
public Action<RegexOptions> enableCompilationFlag(RegexOptions options) => return options | RegexOptions.Compiled;
public Action<RegexOptions> noCompilationFlag(RegexOptions options) => return options;

public static IEnumerable<object[]> Split_NonCompiled_TestData(Action<RegexOptions> transform)
{
      ....
yield return new object[] { "[abc]", "1A2B3C4", transform(RegexOptions.IgnoreCase), 7, 0, new string[] { "1", "2", "3", "4" } };
...
}
```

Would something like that work?

Reply:
Or maybe something else you can devise that doesn't need reflection.

Reply:
It would definitely work but I'm not a huge fan of changing the existing test data for that. I think my current approach has the advantage that you can "just enable" it. Of course it uses reflection but I don't see any issue with that in test code. It would even work in uap/uapaot if we would like that.

That said I improved documentation of that method and removed unused code, I think it's now relatively simple to understand.

Reply:
@stephentoub preferences?